core/store/hashtree/binprefix/tree.go

```diff
@@ -103,6 +103,73 @@ func (t *Tree) Len() int {
 	return counter
 }
 
+// Load scans the bucket for leafs to insert them in the tree. It will then
+// persist the tree to restore the memory depth limit.
+func (t *Tree) Load(bucket kv.Bucket) error {
```

Comment:
less elegant, but easier to understand
```suggestion
func (t *Tree) FillFrom(bucket kv.Bucket) error {
```

Reply:
No that's actually a good name.